pkg/cache/v3/simple.go

```diff
@@ -419,7 +432,13 @@ func (cache *snapshotCache) CreateDeltaWatch(request *DeltaRequest, state stream
 			cache.log.Errorf("failed to compute version for snapshot resources inline, waiting for next snapshot update")
 			delayedResponse = true
 		}
-		delayedResponse = respondDelta(request, value, state, snapshot, cache.log) == nil
+		response, err := respondDelta(context.Background(), request, value, state, snapshot, cache.log)
+		if err != nil {
+			cache.log.Errorf("failed to respond with delta response, waiting for next snapshot update: %s", err)
```

Comment:
We usually check for `cache.log == nil` in other places, this could result in a panic.

Reply:
Good call, let me address this when I get a chance (if nobody else gets to it, I'm about done for the week) and also add a test for a nil logger

Reply:
I'll open this quickly, I missed that in the review sorry about that